src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/LinearAVF.java

```diff
@@ -292,360 +192,49 @@ public double getWeight(Criterion crit) {
-  }
-
-  /**
-   * Set the function which will be used to calculate the subjective value of the accommodation
-   * capacity.
-   *
-   * @param nbSleepingValueFunction the new LinearValueFunction for NbSleeping
-   */
-  private void setNbSleepingValueFunction(LinearValueFunction nbSleepingValueFunction) {
-    this.nbSleepingValueFunction = checkNotNull(nbSleepingValueFunction);
-    LOGGER.info("The number of sleeping preferencies has been set");
-  }
-
-  /**
-   * Gets the object PartialValueFunction used to compute the subjective value of the price per
-   * night
-   *
-   * @return the attribute pricePerNightValueFunction
-   */
-  public ReversedLinearValueFunction getPricePerNightValueFunction() {
-    return this.pricePerNightValueFunction;
-  }
-
-  /**
-   * Set the function which will be used to calculate the subjective value of the price per night.
-   *
-   * @param pricePerNightValueFunction the new ReversedLinearValueFunction for PricePerNight
-   */
-  private void setPricePerNightValueFunction(
-      ReversedLinearValueFunction pricePerNightValueFunction) {
-    this.pricePerNightValueFunction = checkNotNull(pricePerNightValueFunction);
-    LOGGER.info("The price per night preferencies has been set");
-  }
-
-  /**
-   * Gets the object PartialValueFunction used to compute the subjective value of the presence of a
-   * television
-   *
-   * @return the attribute teleValueFunction
-   */
-  public BooleanValueFunction getTeleValueFunction() {
-    return this.teleValueFunction;
+  private void setInternalValueFunction(Criterion criterion) {
+    if (criterion.isNotBooleanCrescent()) {
+      this.linearValueFunctions.put(criterion, null);
+    } else if (criterion.isNotBooleanDecrease()) {
+      this.reversedValueFunctions.put(criterion, null);
+    } else {
+      throw new IllegalArgumentException(
+          "Criterion must be IS_NOT_BOOLEAN_CRESCENT or IS_NOT_BOOLEAN_DECREASE");
+    }
   }
 
-  /**
-   * Set the function which will be used to calculate the subjective value of the presence of a
-   * television.
-   *
-   * @param teleValueFunction the new BooleanValueFunction for Tele
-   */
-  private void setTeleValueFunction(BooleanValueFunction teleValueFunction) {
-    this.teleValueFunction = checkNotNull(teleValueFunction);
-    LOGGER.info("The tele preferencies has been set");
+  private void setInternalValueFunction(Criterion criterion, LinearValueFunction vf) {
+    checkArgument(criterion.isNotBooleanCrescent());
+    this.linearValueFunctions.put(criterion, vf);
   }
 
-  /**
-   * Gets the object PartialValueFunction used to compute the subjective value of the presence of a
-   * terrace
-   *
-   * @return the attribute terraceValueFunction
-   */
-  public BooleanValueFunction getTerraceValueFunction() {
-    return this.terraceValueFunction;
+  private void setInternalValueFunction(Criterion criterion, ReversedLinearValueFunction vf) {
+    checkArgument(criterion.isNotBooleanDecrease());
```

Comment:
Idem checkNotNull svp

Reply:
Fix in d34a55ad54fb6e348608d341481c740cbb7ff702